File: Prime-generator/Javascript/3.js
function is_prime(n) {
    if (n==2) {
        return true
    }

    if (n%2==0) {
        return false
    }

    for (let i=3; i<Math.sqrt(n); i+=2) {
        if (n % i == 0)
            return false
    }
    return true
}



const start = process.hrtime.bigint();
const primes = [];
for (let i=2; i<process.argv[2]; i++) {
    if (is_prime(i)) {
        primes.push(i)
    }
}
const end = process.hrtime.bigint();
const duration = end - start;
console.log(`Execution time: ${duration} ns`);

File: Prime-generator/Python/3.py
import sys
import time
from math import sqrt


def is_prime(n):
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(sqrt(n)), 2):
        if n % i == 0:
            return False
    return True




start = time.time_ns()
primes = []
for i in range(2, int(sys.argv[1])):
    if is_prime(i):
        primes.append(i)
duration = time.time_ns() - start
print(f"Execution time: {duration} ns")